### PR TITLE
remove apiToken from log object

### PR DESCRIPTION
**Pull request by daniel-greening**
### Problem Description
- Okta publish sensitive apiToken information in log objects 

### Solution Description
- Remove this field from the object for the time being

### CODE REVIEW

File: collectors/okta/okta_collector.js [hunk 1]
```diff
@@ -95,6 +95,9 @@ class OktaCollector extends PawsCollector {
         const ts = parse.getMsgTs(msg, tsPaths);
         const typeId = parse.getMsgTypeId(msg, typeIdPaths);
 
+        // TODO remove this when okta filter this field out
+        delete msg.client.apiToken;
```
Comment (kkuzmin) [inline collectors/okta/okta_collector.js]: what happens if `msg.client` property doesn't exist?

Comment (OHaganA) [reply collectors/okta/okta_collector.js]: Delete won't have any effect and just returns `true`

Comment (JDragovichAlertLogic) [reply collectors/okta/okta_collector.js]: The api token is not just in that spot. It needs to be removed in the default http headers.

```
 "apiToken": "WHYUDOTHISOKTA?",
        "http": {
            "defaultHeaders": {
                "Authorization": "SSWS WHYUDOTHISOKTA?",
                "User-Agent": "@okta/okta-sdk-nodejs/3.1.0 node/11.10.0 darwin/18.7.0"
            },
```

File: collectors/okta/okta_collector.js [hunk 2]
```diff
@@ -92,9 +97,14 @@ class OktaCollector extends PawsCollector {
     pawsFormatLog(msg) {
         let collector = this;
 
+        console.log("message = ", msg);
```
Comment (OHaganA) [inline collectors/okta/okta_collector.js]: Stray console log here

File: collectors/okta/okta_collector.js [hunk 3]
```diff
@@ -112,6 +122,21 @@ class OktaCollector extends PawsCollector {
...
+        let properties = Array.isArray(propertyPath) ? propertyPath : propertyPath.split(".")
+
+        if (properties.length > 1) {
+            if (!obj.hasOwnProperty(properties[0]))
+                return true
+
+            return this.redactValue(properties.slice(1), obj[properties[0]])
+        } else {
+            obj[properties[0]] = undefined
+            console.log("object", obj)
```
Comment (OHaganA) [inline collectors/okta/okta_collector.js]: Same here with the console log

**Pull request merged**